packages/itwin/grouping-mapping-widget/src/test/QueryBuilder.testdata.json

```diff
@@ -0,0 +1,808 @@
+            }
+          },
+          "propertiesField": {
+            "properties": [
+              {
+                "property": {
+                  "name": "somePropName",
+                  "classInfo": {
+                    "name": "A:B"
+                  }
+                }
+              }
+            ],
+            "type": {
+              "typeName": "number"
+            },
+            "parent": {
+              "pathToPrimaryClass": []
+            }
+          }
+        }
+      ]
+    },
+    {
+      "name": "when relational property added, return query string with property value and relation chain to it",
+      "expectedResult": "SELECT Z.X.ECInstanceId FROM Z.X JOIN A.E ON A.E.SourceECInstanceId=A.C.ECInstanceId JOIN A.C ON A.C.ECInstanceId=A.D.SourceECInstanceId JOIN A.D ON A.D.TargetECInstanceId=A.B.ECInstanceId JOIN A.B ON A.B.somePropName='someString' WHERE Z.X.ECInstanceId=A.E.TargetECInstanceId",
+      "operations": [
+        {
+          "expectedResult": true,
+          "operationType": "addProperty",
+          "propertyRecord": {
+            "value": {
+              "valueFormat": 0,
+              "value": "someString"
+            },
+            "property": {
+              "typename": "notnavigation"
+            }
+          },
+          "propertiesField": {
+            "properties": [
+              {
+                "property": {
+                  "name": "somePropName",
+                  "classInfo": {
+                    "name": "A:B"
+                  }
+                }
+              }
+            ],
+            "type": {
+              "typeName": "string"
+            },
+            "parent": {
+              "pathToPrimaryClass": [
+                {
+                  "sourceClassInfo": {
+                    "id": "id1",
+                    "name": "A:B",
+                    "label": "AB label"
+                  },
+                  "targetClassInfo": {
+                    "id": "id2",
+                    "name": "A:C",
+                    "label": "AC label"
+                  },
+                  "isPolymorphicTargetClass": true,
+                  "relationshipInfo": {
+                    "id": "id3",
+                    "name": "A:D",
+                    "label": "AD label"
+                  },
+                  "isPolymorphicRelationship": true,
+                  "isForwardRelationship": false
+                },
+                {
+                  "sourceClassInfo": {
+                    "id": "id2",
+                    "name": "A:C",
+                    "label": "AC label"
+                  },
+                  "targetClassInfo": {
+                    "id": "id4",
+                    "name": "Z:X",
+                    "label": "ZX label"
+                  },
+                  "isPolymorphicTargetClass": false,
+                  "relationshipInfo": {
+                    "id": "id5",
+                    "name": "A:E",
```

Comment:
It's pretty hard to review this. Maybe prefixing Relationship class names with `Rel` would help a bit.

Reply:
Changed

Reply:
confirmed